primitives/src/conststr.rs

```diff
@@ -453,6 +469,138 @@ impl TryFrom<String> for InlinedString {
     }
 }
 
+mod ffi {
+    #![allow(unsafe_code)]
+    use core::mem::ManuallyDrop;
+
+    use iroha_ffi::{
+        ir::{Ir, Transmute},
+        repr_c::{COutPtr, CType, CTypeConvert},
+        slice::OutBoxedSlice,
+    };
+
+    use super::*;
+
+    #[repr(C)]
+    #[derive(Clone, Copy)]
+    pub union ReprCConstString {
+        inlined: InlinedString,
+        boxed: ReprCBoxedString,
+    }
+
+    #[repr(C)]
+    #[derive(Debug, Clone, Copy)]
+    pub struct ReprCBoxedString {
```

Comment:
This made me think whatewer we should use `*mut u8` in `BoxedString` instead of `NonNull<u8>`.

Reply:
not sure if it would make any difference to the `rustc`?

Reply:
`ReprC` structs are required to be `Copy` currently. The choice is either to remove the `Copy` supertrait and apply the suggestion or leave it as is. after discussion with @Erigara it was decided to leave it as is for the moment at least and possibly revisit in the future.